confuse.py

```diff
@@ -1183,6 +1183,32 @@ def __repr__(self):
         return 'MappingTemplate({0})'.format(repr(self.subtemplates))
 
 
+class SequenceTemplate(Template):
+    """A template used to validate lists of similar items.
+    Note that only the first item of the template list is used to
+    validate against.
+    """
+    def __init__(self, sequence):
+        """Create a template based on the first item in list.
+        If given an empty list, assume the validated list must be empty.
+        """
```

Comment:
Instead of initializing this with a sequence, I recommend making the argument just the wrapped sub-template (i.e., `sequence[0]` here). I think this makes the behavior of the sequence template itself a little clearer—it doesn't need to enforce empty lists, doesn't have to ignore lists with length greater than 1, and generally behaves more like a "combinator" that builds a larger template out of a smaller one.

Reply:
I think so! Unless I'm misunderstanding, the Sequence template recognizes a list *of any length* whose members all match the underlying template. Is that possible to accomplish somehow using existing templates?

Reply:
Please take a look at my latest commit. https://github.com/sampsyo/confuse/pull/43/commits/8aeae3ca9f5a509a14a3d5e7a4009262f6c597cf
The Sequence constructor takes the subtemplate as its argument. Then the value method returns the list of values of that subtemplate iterated over each item of the view. Which shold be iterable, an exception is raised if it's not. I checked, but I really should write another unit test for this case.